### ISSUE TITLE
Tests failing dues to incorrect escaping of RE strings

### ISSUE THREAD

**Issue by mcepl** [aio-libs/aiohttp#3325]
## Long story short

```
[   53s] platform linux -- Python 3.6.5, pytest-3.8.1, py-1.5.4, pluggy-0.7.1
[   53s] rootdir: /home/abuild/rpmbuild/BUILD/aiohttp-3.4.4, inifile: setup.cfg
[   53s] plugins: timeout-1.3.2, mock-1.10.0
...
build/rpmbuild/BUILD/aiohttp-3.4.4/aiohttp/_http_writer.cpython-36m-x86_64-linux-gnu.so
[   70s] ============================= test session starts ==============================
[   70s] platform linux -- Python 3.6.5, pytest-3.8.1, py-1.5.4, pluggy-0.7.1
[   70s] rootdir: /home/abuild/rpmbuild/BUILD/aiohttp-3.4.4, inifile: setup.cfg
[   70s] plugins: timeout-1.3.2, mock-1.10.0
[   77s] collected 1910 items / 4 errors
[   77s] 
[   77s] ==================================== ERRORS ====================================
[   77s] __________________ ERROR collecting tests/test_http_parser.py __________________
[   77s] /usr/lib/python3.6/site-packages/_pytest/python.py:482: in _importtestmodule
[   77s]     mod = self.fspath.pyimport(ensuresyspath=importmode)
[   77s] /usr/lib/python3.6/site-packages/py/_path/local.py:668: in pyimport
[   77s]     __import__(modname)
[   77s] E     File "/home/abuild/rpmbuild/BUILD/aiohttp-3.4.4/tests/test_http_parser.py", line 372
[   77s] E       match = ("400, message='Got more than 8190 bytes \({}\) when reading"
[   77s] E               ^
[   77s] E   SyntaxError: invalid escape sequence \(
[   77s] ____________________ ERROR collecting tests/test_streams.py ____________________
[   77s] /usr/lib/python3.6/site-packages/_pytest/python.py:482: in _importtestmodule
[   77s]     mod = self.fspath.pyimport(ensuresyspath=importmode)
[   77s] /usr/lib/python3.6/site-packages/py/_path/local.py:668: in pyimport
[   77s]     __import__(modname)
[   77s] E     File "/home/abuild/rpmbuild/BUILD/aiohttp-3.4.4/tests/test_streams.py", line 724
[   77s] E       assert re.search("<StreamReader w=<Future pending[\S ]*>>",
[   77s] E                       ^
[   77s] E   SyntaxError: invalid escape sequence \S
[   77s] __________________ ERROR collecting tests/test_urldispatch.py __________________
[   77s] /usr/lib/python3.6/site-packages/_pytest/python.py:482: in _importtestmodule
[   77s]     mod = self.fspath.pyimport(ensuresyspath=importmode)
[   77s] /usr/lib/python3.6/site-packages/py/_path/local.py:668: in pyimport
[   77s]     __import__(modname)
[   77s] E     File "/home/abuild/rpmbuild/BUILD/aiohttp-3.4.4/tests/test_urldispatch.py", line 589
[   77s] E       route = router.add_route('GET', '/get/{num:^\d+}', handler,
[   77s] E                                      ^
[   77s] E   SyntaxError: invalid escape sequence \d
[   77s] __________________ ERROR collecting tests/test_web_request.py __________________
[   77s] /usr/lib/python3.6/site-packages/_pytest/python.py:482: in _importtestmodule
[   77s]     mod = self.fspath.pyimport(ensuresyspath=importmode)
[   77s] /usr/lib/python3.6/site-packages/py/_path/local.py:668: in pyimport
[   77s]     __import__(modname)
[   77s] E     File "/home/abuild/rpmbuild/BUILD/aiohttp-3.4.4/tests/test_web_request.py", line 343
[   77s] E       header = 'BY=identifier;pROTO="\lala lan\d\~ 123\!&"'
[   77s] E               ^
[   77s] E   SyntaxError: invalid escape sequence \l
[   77s] !!!!!!!!!!!!!!!!!!! Interrupted: 4 errors during collection !!!!!!!!!!!!!!!!!!!!
[   77s] =========================== 4 error in 7.42 seconds ============================
[   78s] error: Bad exit status from /var/tmp/rpm-tmp.2YgiWK (%check)
[   78s] 
[   78s] 
[   78s] RPM build errors:
[   78s]     Bad exit status from /var/tmp/rpm-tmp.2YgiWK (%check)
[   78s] 
[   78s] stitny failed "build python-aiohttp.spec" at Sat Oct  6 08:56:18 UTC 2018.
[   78s] 
```

## Expected behaviour

Test suite should pass without problems.

## Actual behaviour

It fails as shown above

## Steps to reproduce

Just run testsuite with ``python3 setup.py test``

## Your environment
```
[  124s] platform linux -- Python 3.6.5, pytest-3.8.1, py-1.5.4, pluggy-0.7.1
```

chroot build environment for openSUSE, details at [SPEC file](https://github.com/aio-libs/aiohttp/files/2452878/python-aiohttp.spec.txt) and full [build log](https://github.com/aio-libs/aiohttp/files/2452879/python-aiohttp.build-log.txt).

This [patch](https://github.com/aio-libs/aiohttp/files/2452880/invalid-escapes-in-tests.patch.txt) fixes the issue.

**Comment by mcepl** [comment]
@webknjaz I have couple of hundred packages to manage (plus I dabble in almost thousand more) as a openSUSE packager. Do I have to clone all of them, just because its upstream maintainers are too lazy to apply a patch file themselves?

**Comment by mcepl** [comment]
> At least post your github email, so that we could use Co-authored-by: in commit message. Please.

You are right, I have overreacted. I am sorry. It is mcepl at cepl dot eu. The name is Matěj Cepl

**Comment by gyermolenko** [comment]
@asvetlov has already fixed it with f3a94ce7c8beeecbb4d5a64d7adb14f4ce30b8c0
It is in master (and not available in latest 3.4.4 release)

**Comment by asvetlov** [comment]
The problem is fixed.
I see no activity in providing a more aesthetic patch.
Closing.
